rxros/include/rxros/rxros.h

```diff
@@ -301,7 +301,7 @@ namespace rxros
         auto publish_to_topic(const std::string &topic, const uint32_t queue_size = 10) {
             return [=](auto&& source) {
                 ros::Publisher publisher(rxros::node::get_handle().advertise<T>(topic, queue_size));
-                source.observe_on(rxcpp::synchronize_new_thread()).subscribe(
+                source.subscribe_on(rxcpp::synchronize_new_thread()).subscribe(
```

Comment:
I have been reading about this in a book and online, and as far as I can see there should be no difference between the two operators in this case. Observe_on immediately spawns a new thread for all subsequent lambda and the subscription (so the subscribed callbacks all called in the parallel thread to the current thread).    Subscribe_on has the same effect only for the subsequent subscription, but not for the subsequent lambdas (map, filter, etc).  In this case we only have the subscribe afterwards, so the effect should be identical.  Have you observed any difference in testing @henrik7264 ?

There is some background on this here: https://stackoverflow.com/questions/44984730/rxandroid-whats-the-difference-between-subscribeon-and-observeon (two top answers are worth reading).

Thinking longer about it, I believe that we should remove observe_on (and subscribe_on), not replace it.  This gives the programmer maximum flexibility.  This way publish_to_topic will not spawn *any* new threads, unless the programmer explicitly asks for it by inserting observe_on or subscribe_on.